types/emscripten/index.d.ts

```diff
@@ -28,96 +25,104 @@ declare namespace Module {
+    UTF16ToString(ptr: number): string;
+    stringToUTF16(str: string, outPtr: number): void;
+    UTF32ToString(ptr: number): string;
+    stringToUTF32(str: string, outPtr: number): void;
 
     // USE_TYPED_ARRAYS == 1
-    var HEAP: Int32Array;
-    var IHEAP: Int32Array;
-    var FHEAP: Float64Array;
+    HEAP: Int32Array;
+    IHEAP: Int32Array;
+    FHEAP: Float64Array;
 
     // USE_TYPED_ARRAYS == 2
-    var HEAP8: Int8Array;
-    var HEAP16: Int16Array;
-    var HEAP32: Int32Array;
-    var HEAPU8:  Uint8Array;
-    var HEAPU16: Uint16Array;
-    var HEAPU32: Uint32Array;
-    var HEAPF32: Float32Array;
-    var HEAPF64: Float64Array;
-
-    var TOTAL_STACK: number;
-    var TOTAL_MEMORY: number;
-    var FAST_MEMORY: number;
-
-    function addOnPreRun(cb: () => any): void;
-    function addOnInit(cb: () => any): void;
-    function addOnPreMain(cb: () => any): void;
-    function addOnExit(cb: () => any): void;
-    function addOnPostRun(cb: () => any): void;
+    HEAP8: Int8Array;
+    HEAP16: Int16Array;
+    HEAP32: Int32Array;
+    HEAPU8:  Uint8Array;
+    HEAPU16: Uint16Array;
+    HEAPU32: Uint32Array;
+    HEAPF32: Float32Array;
+    HEAPF64: Float64Array;
+
+    TOTAL_STACK: number;
+    TOTAL_MEMORY: number;
+    FAST_MEMORY: number;
+
+    addOnPreRun(cb: () => any): void;
+    addOnInit(cb: () => any): void;
+    addOnPreMain(cb: () => any): void;
+    addOnExit(cb: () => any): void;
+    addOnPostRun(cb: () => any): void;
 
     // Tools
-    function intArrayFromString(stringy: string, dontAddNull?: boolean, length?: number): number[];
-    function intArrayToString(array: number[]): string;
-    function writeStringToMemory(str: string, buffer: number, dontAddNull: boolean): void;
-    function writeArrayToMemory(array: number[], buffer: number): void;
-    function writeAsciiToMemory(str: string, buffer: number, dontAddNull: boolean): void;
+    intArrayFromString(stringy: string, dontAddNull?: boolean, length?: number): number[];
+    intArrayToString(array: number[]): string;
+    writeStringToMemory(str: string, buffer: number, dontAddNull: boolean): void;
+    writeArrayToMemory(array: number[], buffer: number): void;
+    writeAsciiToMemory(str: string, buffer: number, dontAddNull: boolean): void;
 
-    function addRunDependency(id: any): void;
-    function removeRunDependency(id: any): void;
+    addRunDependency(id: any): void;
+    removeRunDependency(id: any): void;
 
 
-    var preloadedImages: any;
-    var preloadedAudios: any;
+    preloadedImages: any;
+    preloadedAudios: any;
 
-    function _malloc(size: number): number;
-    function _free(ptr: number): void;
+    _malloc(size: number): number;
+    _free(ptr: number): void;
 }
 
+// By default Emscripten emits a single global Module.  Users setting -s
+// MODULARIZE=1 -s EXPORT_NAME=MyMod should declare their own types, e.g.
+// declare var MyMod: EmscriptenModule;
+declare var Module: EmscriptenModule;
```

Comment:
Maybe we could omit this one, so that every person defines the correct module name. If one uses the default settings, Emscripten will indeed emit `Module`, but otherwise this will "pollute" the global types. What do you think?

Reply:
The default Emscripten configuration uses the global Module name.  It's unfortunate that it's so generic but I don't know of any specific conflicts with it.  @periklis in the linked issue (#24791) claims that the MODULARIZE option with other names is not much used.

Of the two use cases I have encountered at work where we have just imported this typings definition, one uses Module and the other sets a name with MODULARIZE.

Reply:
I'd prefer to keep the annotation in order to make using the typings definition simple for Emscripten users with the default set up.